packages/gatsby-image/src/index.js

```diff
@@ -100,11 +100,34 @@ const noscriptImg = props => {
     ? `crossorigin="${props.crossOrigin}" `
     : ``
 
-  return `<picture>${srcSetWebp}<img ${width}${height}${sizes}${srcSet}${src}${alt}${title}${crossOrigin}style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/></picture>`
+  // Since we're in the noscript block for this image (which is rendered during SSR or when js is disabled),
+  // we have no way to "detect" if native lazy loading is supported by the user's browser (at least not today — See 1)
+  // Since this attribute is a progressive enhancement, it won't break a browser with no support
+  // Therefore setting it by default is a good idea.
+
+  // 1. In the future, a Client Hint for this might be available (https://bugs.chromium.org/p/chromium/issues/detail?id=949365)
+
+  const loading = props.loading ? `loading="${props.loading}" ` : ``
+
+  return `<picture>${srcSetWebp}<img ${loading}${width}${height}${sizes}${srcSet}${src}${alt}${title}${crossOrigin}style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/></picture>`
 }
 
 const Img = React.forwardRef((props, ref) => {
-  const { sizes, srcSet, src, style, onLoad, onError, ...otherProps } = props
+  const {
+    sizes,
+    srcSet,
+    src,
+    style,
+    onLoad,
+    onError,
+    nativeLazyLoadSupported,
+    loading,
+    ...otherProps
+  } = props
+
+  const loadingAttribute = {
+    ...(nativeLazyLoadSupported && loading),
```

Comment:
Is `nativeLazyLoadSupported` boolean actually useful here?

This doesn't seem like the right approach for handling this single attribute. You're defining a const obj with a potential single prop that if the boolean prop is true, adopts the `loading` var as a key/prop to it's contained string value?

The destructuring approach here and below for this seems to harm readability. Why not instead go with:

```JS
loading={loading || `auto`}
// or conditionally, if the prop isn't specified, the attribute won't be assigned
loading={loading}
```

If the browser supports this attribute, `auto` is applied by default implicitly, there isn't a need to check for native support to assign it, makes no difference.